core/src/serve_repair.rs

```diff
@@ -432,19 +454,37 @@ impl ServeRepair {
         packet_batch: PacketBatch,
         response_sender: &PacketBatchSender,
         stats: &mut ServeRepairStats,
+        data_budget: &DataBudget,
     ) {
         // iter over the packets
-        packet_batch.iter().for_each(|packet| {
+        let mut last_processed_packet_index = None;
+        for (i, packet) in packet_batch.iter().enumerate() {
+            last_processed_packet_index = Some(i);
             if let Ok(request) = packet.deserialize_slice(..) {
                 stats.processed += 1;
                 let from_addr = packet.meta.socket_addr();
                 let rsp = Self::handle_repair(me, recycler, &from_addr, blockstore, request, stats);
-                stats.total_response_packets += rsp.as_ref().map(PacketBatch::len).unwrap_or(0);
+                let num_response_packets = rsp.as_ref().map(PacketBatch::len).unwrap_or(0);
                 if let Some(rsp) = rsp {
```

Comment:
Can we simplify this code to:
```rust
let rsp = match Self::handle_repair(me, recycler, &from_addr, blockstore, request, stats) {
  None => continue,
  Some(rsp) => rsp,
};
```

Won't need `.as-ref().map(...).unwrap_or(...)` any more.